src/file-systems/base.ts

```diff
@@ -18,8 +18,9 @@ interface Params {
     lstatSync(path: string): {
       isFile(): boolean;
     };
+    openSync(path: string, flags: string): number;
```

Comment:
Replace `flags: string` by only the allowed flags, so far:

```suggestion
    openSync(path: string, flags: "r"): number;
```